service/history/execution/mutable_state_util_test.go

```diff
@@ -316,3 +327,252 @@ func TestConvertWorkflowRequests(t *testing.T) {
 
 	assert.ElementsMatch(t, expectedOutputs, convertWorkflowRequests(inputs))
 }
+
+func TestCreatePersistenceMutableState(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockShardContext := shard.NewMockContext(ctrl)
+	logger := testlogger.New(t)
+	mockEventsCache := events.NewMockCache(ctrl)
+	mockDomainCache := cache.NewMockDomainCache(ctrl)
+	mockShardContext.EXPECT().GetClusterMetadata().Return(constants.TestClusterMetadata).Times(2)
+	mockShardContext.EXPECT().GetEventsCache().Return(mockEventsCache)
+	mockShardContext.EXPECT().GetConfig().Return(config.NewForTest())
+	mockShardContext.EXPECT().GetTimeSource().Return(clock.NewMockedTimeSource())
+	mockShardContext.EXPECT().GetMetricsClient().Return(metrics.NewClient(tally.NoopScope, metrics.History))
+	mockShardContext.EXPECT().GetDomainCache().Return(mockDomainCache)
+
+	builder := newMutableStateBuilder(mockShardContext, logger, constants.TestLocalDomainEntry)
+	builder.pendingActivityInfoIDs[0] = &persistence.ActivityInfo{}
+	builder.pendingTimerInfoIDs["some-key"] = &persistence.TimerInfo{}
+	builder.pendingSignalInfoIDs[0] = &persistence.SignalInfo{}
+	builder.pendingChildExecutionInfoIDs[0] = &persistence.ChildExecutionInfo{}
+	builder.bufferedEvents = []*types.HistoryEvent{{}}
+	builder.updateBufferedEvents = []*types.HistoryEvent{{}}
+	builder.versionHistories = &persistence.VersionHistories{CurrentVersionHistoryIndex: 0, Histories: []*persistence.VersionHistory{}}
+	builder.pendingRequestCancelInfoIDs[0] = &persistence.RequestCancelInfo{}
+	builder.decisionTaskManager.(*mutableStateDecisionTaskManagerImpl).HasInFlightDecision()
+	builder.executionInfo.DecisionStartedID = 1
+
+	mutableState := CreatePersistenceMutableState(builder)
+	assert.NotNil(t, mutableState)
```

Comment:
nit: is it possible to check content of the mutableState? For instance, to make sure the fields are really copied from builder.